RecoMET/METAlgorithms/src/METSignificance.cc

```diff
@@ -0,0 +1,184 @@
+// -*- C++ -*-
+//
+// Package:    METAlgorithms
+// Class:      METSignificance
+// 
+/**\class METSignificance METSignificance.cc RecoMET/METAlgorithms/src/METSignificance.cc
+Description: [one line class summary]
+Implementation:
+[Notes on implementation]
+*/
+//
+// Original Author:  Nathan Mirman (Cornell University)
+//         Created:  Thu May 30 16:39:52 CDT 2013
+//
+//
+
+#include "RecoMET/METAlgorithms/interface/METSignificance.h"
+
+
+metsig::METSignificance::METSignificance(const edm::ParameterSet& iConfig) {
+
+  edm::ParameterSet cfgParams = iConfig.getParameter<edm::ParameterSet>("parameters");
+
+
+  std::string ptResFileName = cfgParams.getParameter<std::string>("ptResFile");
+  std::string phiResFileName = cfgParams.getParameter<std::string>("phiResFile");
+
+  double dRmatch = cfgParams.getParameter<double>("dRMatch");
+  dR2match_ = dRmatch*dRmatch;
+  
+  jetThreshold_ = cfgParams.getParameter<double>("jetThreshold");
+  jetEtas_ = cfgParams.getParameter<std::vector<double> >("jeta");
+  jetParams_ = cfgParams.getParameter<std::vector<double> >("jpar");
+  pjetParams_ = cfgParams.getParameter<std::vector<double> >("pjpar");
+
+
+  edm::FileInPath fpt("CondFormats/JetMETObjects/data/"+ptResFileName);
+  edm::FileInPath fphi("CondFormats/JetMETObjects/data/"+phiResFileName);
+
+  ptRes_  = new JetResolution(fpt.fullPath().c_str(),false);
+  phiRes_ = new JetResolution(fphi.fullPath().c_str(),false);
+
+  fPtEta_ = nullptr;
+  fPhiEta_ = nullptr;
+
+}
+
+metsig::METSignificance::~METSignificance() {
+  delete ptRes_;
+  delete phiRes_;
+  delete fPtEta_;
```

Comment:
there is one delete in the ::getCovariance call
better use smart pointers (shared_ptr or unique_ptr)